src/plots/cartesian/dragbox.js

```diff
@@ -168,13 +171,31 @@ module.exports = function dragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             else if(isSelectOrLasso(dragModeNow)) {
                 dragOptions.xaxes = xa;
                 dragOptions.yaxes = ya;
+
+                // take over selection polygons from prev mode, if any
+                if((e.shiftKey || e.altKey) && (plotinfo.selection && plotinfo.selection.polygons) && !dragOptions.polygons) {
+                    dragOptions.polygons = plotinfo.selection.polygons;
+                    dragOptions.mergedPolygons = plotinfo.selection.mergedPolygons;
+                }
+                // create new polygons, if shift mode
+                else if((!e.shiftKey && !e.altKey) || ((e.shiftKey || e.altKey) && !plotinfo.selection)) {
+                    plotinfo.selection = {};
+                    plotinfo.selection.polygons = dragOptions.polygons = [];
+                    plotinfo.selection.mergedPolygons = dragOptions.mergedPolygons = [];
+                }
+
                 prepSelect(e, startX, startY, dragOptions, dragModeNow);
             }
         }
     };
 
     dragElement.init(dragOptions);
 
+    // FIXME: this hack highlights selection once we enter select/lasso mode
+    if(isSelectOrLasso(gd._fullLayout.dragmode) && plotinfo.selection) {
```

Comment:
I'm not a big fan of this behavior. @dfcreative is there a particular reason why you added this?

Reply:
Not relevant since #2135